interview-questions.md

```diff
@@ -7,65 +7,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+  Your answer: Post, Get, Put & Patch, and Delete. These are methods to enact HTTP's form of C.R.U.D operations, which stands for Create, Read, Update, and Delete.. These are the essential dynamics of creating a persistent storage system, or creating a website that is capable of holding & manipulating data. So, Post would refer to Creating a website, Get would refer to Reading, Put and Patch refer to Updating, and Delete is to Delete/Destroying.
 
-  Researched answer:
+  Researched answer: Post, get, put & patch, and delete are the minimal necessary methods that are required to create a full stack web application. They are used to Create, Read, Update, and Delete data from a persistent storage system that a web application is using. It is the only way to manipulate and handle the data a website deals with.
 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: This is a tool used to interact with incompatible data types using Object-Oriented programming languages. So, in Rails, Active Record is an ORM tool used convert data types and interact with them since they are incompatible and the only way to convert data between incompatible systems.
 
-  Researched answer:
+  Researched answer: Object-raltional mapping is a tool that developers use to interact and convert data between incompatible systems using Object-Oriendted programming languages. It enables developers to work with data that is incompatible with each other.
 
 
 
 6. What is a gem?
 
-  Your answer:
+  Your answer: Gems essentially are Ruby applications that help extend the capabilities of Ruby, so you can have a gem that runs tests for you like RSPEC.
 
-  Researched answer:
+  Researched answer: Gems are a library of code functionality. So they package up Ruby code to make it simple to share with others and they help extend or modify the functionality of ruby applications.
 
 
 
 7. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are sort of like ID tags that are used to help identify every row of data. So they are sort of like unique identifiers that help pinpoint exact data sets inside a database. They are important because there could be many first and last names in a data base, but they are all associated with a single primary key to refer to.
 
-  Researched answer:
+  Researched answer: Primary keys are important because they are unique identifiers designated to identify each table record. They are important because sets of data can be assigned a single row and all pieces of data that are related to each other can all be identified by a primary key. They are like social security numbers in the United States, so with my name Ryan Sarll, or with another name like Jose Gonzales, there is probably more than one person in the US named our names, so we are given SSN's, or Primary Keys, to refer to us as individuals that contain all our relevant data.
```

Comment:
I really like that you provided an example of a primary key and the significance for how it's used! Keep this up!